client/packages/common/src/ui/layout/tables/DataTable.tsx

```diff
@@ -180,12 +183,6 @@ const DataTableComponent = <T extends RecordWithId>({
     if (page * first > total) onChangePage(0);
   }, [pagination]);
 
-  // if the columns array changes, such as when a plugin column is added
-  // reset the display columns to the full set
```

Comment:
No need to reset all columns now -- the new one appears and all other columns retain their visibility status.